src/formMapper.js

```diff
@@ -0,0 +1,62 @@
+const formElSelector = 'form';
+const multiSelectSelector = 'select[multiple]';
+const radioSelector = 'input[type="radio"]';
+const checkboxSelector = 'input[type="checkbox"]';
+const defaultElementSelectors = 'input, select, textarea';
+
+function getInputElements(formEl, selectorSettings = defaultElementSelectors) {
+    return [...formEl.querySelectorAll(selectorSettings)];
+}
+
+function mapValues(elementList) {
+    const returnVal = {};
+    for (const el of elementList) {
+        if (!el.name) {
+            continue;
+        }
+
+        if (el.matches(checkboxSelector)) {
+            if (el.checked) {
+                const currentVal = returnVal[el.name] || [];
+                returnVal[el.name] = [...currentVal, el.value];
+            }
+        } else if (el.matches(radioSelector)) {
+            if (el.checked) {
+                returnVal[el.name] = el.value;
+            }
+        } else if (el.matches(multiSelectSelector)) {
+            returnVal[el.name] = [...el.options]
+                                    .filter(optEl => optEl.selected)
+                                    .map(opt => opt.value);
+        } else {
+            returnVal[el.name] = el.value;
+        }
+    }
+    return returnVal;
+}
+
+/**
+ *
+ * @param {Element | Element[]} input a form element or an array of Elements
+ * @param {String} elementSelectors overrides for defaultElementSelectors -- if selectors should be class based, etc
+ * @returns {Object} with the key value pairs being { inputName: inputValue }
+ */
+function getValues(input, elementSelectors = defaultElementSelectors) {
+    if (input) {
+        if (input.tagName) {
+            const elementList = input.tagName.toLowerCase() === formElSelector ? getInputElements(input, elementSelectors) : [input];
```

Comment:
If I pass a container element (that has one or more input elements as children) as the first argument of the getValues() function, this line of code is still requiring that the container element be a form.
This might work:
Scenario #1a: The first argument is a single input element (input/textarea/select). Ignore any second argument and just map the value of that element. 
Scenario #1b: The first argument is a single non-input element.  Assume it's a container element and querySelectAll() the input elements out of it using the second argument. Map the values from those elements.
Scenario #2: The first argument is an array. Assume it's an array of input elements and map the values from those elements.